client-tools/Shell/RequestFuzzer.h

```diff
@@ -0,0 +1,172 @@
+///
+/// @author Julia Puget
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <array>
+#include <charconv>
+#include <optional>
+#include <random>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "Basics/VelocyPackHelper.h"
+#include <fuerte/message.h>
+#include <Rest/CommonDefines.h>
+
+namespace arangodb {
+namespace fuzzer {
+
+enum CharOperation { kAddString = 0, kAddInt32, kMaxCharOpValue };
+
+enum LineOperation {
+  kCopyLine = 0,
+  kInjectRandByteInLine,
+  kAddLine,
+  kMaxLineOpValue
+};
+
+enum BodyOperation { kAddArray = 0, kAddObject, kAddCharSeq, kMaxBodyOpValue };
+
+class RequestFuzzer {
+  struct RandContext {
+    explicit RandContext(uint32_t seed) : mt(seed) {}
+
+    std::mt19937 mt;
+    static constexpr uint32_t maxRandAsciiStringLength = 50;
+  };
+
+  struct Result {
+    std::string restVerb;
+    std::string location;
+  };
+
+ public:
+  RequestFuzzer(std::optional<uint32_t> numIt, std::optional<uint32_t> seed)
+      : _numIterations(numIt.value_or(limitNumIterations)),
+        _seed(seed.value_or(std::random_device()())),
+        _randContext{_seed},
+        _headerSplitInLines{},
+        _keysAndValues{} {}
+
+  uint32_t getSeed() { return _seed; }
+
+  std::unique_ptr<fuerte::Request> createRequest();
+
+ private:
+  void generateHeader(std::string& header);
+
+  void generateBody(velocypack::Builder& builder);
+
+  void randomizeCharOperation(std::string& input, uint32_t numIts);
+
+  void randomizeLineOperation(uint32_t numIts);
+
+  template<typename T>
+  T generateRandNumWithinRange(T min, T max);
+
+  void generateRandAlphaNumericString(std::string& input);
+  void generateRandAsciiChar(std::string& input);
+  void generateRandAlphaNumericChar(std::string& input);
+
+  int32_t generateRandInt32();
+
+  static constexpr size_t limitNumIterations = 10;
+  uint32_t _numIterations;
+  uint32_t _seed;
+  RandContext _randContext;
+  std::vector<std::string> _headerSplitInLines;
+  std::unordered_map<std::string, std::string> _keysAndValues;
+  std::string _tempStr;
+  rest::RequestType _randReqType;
+
+  static constexpr uint32_t kMaxNestedRoutes = 4;
+  static constexpr uint32_t kMaxDepth = 4;
+  static constexpr uint32_t kObjNumMembers = 4;
+  static constexpr uint32_t kArrayNumMembers = 4;
+  static constexpr char _alphaNumericChars[] =
+      "0123456789"
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+      "abcdefghijklmnopqrstuvwxyz";
+
+  uint32_t _recursionDepth = 0;
+  std::vector<std::unordered_set<std::string>> _tempObjectKeys;
+  std::unordered_set<std::string> _usedKeys;
+
+  static constexpr std::array<std::string_view, 12> _wordListForRoute = {
+      {"/_db", "/_admin", "/_api", "/_system", "/_cursor", "/version",
+       "/status", "/license", "/collection", "/database", "/current", "/log"}};
+
+  static constexpr std::array<std::string_view, 48> _wordListForKeys = {
```

Comment:
Can this be moved to the .cpp file?
```suggestion
  static constexpr std::array<std::string_view, 48> wordListForKeys = {
```

Reply:
Addressed